src/material/toolbar/testing/shared.spec.ts

```diff
@@ -0,0 +1,91 @@
+import {Component} from '@angular/core';
+import {ComponentHarness, HarnessLoader} from '@angular/cdk/testing';
+import {MatToolbarModule} from '@angular/material/toolbar';
+import {
+  MatToolbarHarness,
+  MatToolbarSection
+} from '@angular/material/toolbar/testing/toolbar-harness';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {TestbedHarnessEnvironment} from '@angular/cdk/testing/testbed';
+
+export function runHarnessTexts(
+  toolbarModule: typeof MatToolbarModule, toolbarHarness: typeof MatToolbarHarness) {
+  let fixture: ComponentFixture<ToolbarHarnessTest>;
+  let loader: HarnessLoader;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [toolbarModule],
+      declarations: [ToolbarHarnessTest],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ToolbarHarnessTest);
+    fixture.detectChanges();
+    loader = TestbedHarnessEnvironment.loader(fixture);
+  });
+
+  it('should find all toolbars', async () => {
+    const toolbars = await loader.getAllHarnesses(toolbarHarness);
+
+    expect(toolbars.length).toBe(2);
+  });
+
+  it('should find toolbar with text', async () => {
+    const toolbars = await loader.getAllHarnesses(toolbarHarness.with({text: 'My App'}));
+
+    expect(toolbars.length).toBe(1);
+    expect(await toolbars[0].hasMultipleRows()).toBeFalse();
+  });
+
+  it('should find toolbar with regex', async () => {
+    const toolbars = await loader.getAllHarnesses(toolbarHarness.with({text: /Row/}));
+
+    expect(toolbars.length).toBe(1);
+    expect(await toolbars[0].hasMultipleRows()).toBeTrue();
+  });
+
+  it('should get toolbar text', async () => {
+    const toolbars = await loader.getAllHarnesses(toolbarHarness);
+
+    expect(await toolbars[0].getRowsAsText()).toEqual(['My App']);
+    expect(await toolbars[1].getRowsAsText()).toEqual([
+      'Row 1',
+      'Row 2 Button 1  Button 2'
+    ]);
+  });
+
+  it('should get harness loaders for toolbar row', async () => {
+    const toolbar = await loader.getHarness(toolbarHarness.with({text: /Button/}));
+    const rowLoaders = await toolbar.getAllChildLoaders(MatToolbarSection.ROW);
+    const row1 = rowLoaders[0] as HarnessLoader;
+    const row1Subcomponents = await row1.getAllHarnesses(DummyHarness) ?? [];
```

Comment:
Is the nullish coalescing needed? I was under the impression `getAllHarnesses` always returns an array.